src/main/scala/cromwell/binding/types/WdlArrayType.scala

```diff
@@ -28,7 +28,8 @@ case class WdlArrayType(memberType: WdlType) extends WdlType {
         case Success(values) => WdlArray(WdlArrayType(memberType), values)
         case Failure(ex) => throw ex
       }
-
+    case wdlArray: WdlArray if memberType.isCoerceableFrom(wdlArray.wdlType.memberType) =>
+      wdlArray.map(v => memberType.coerceRawValue(v).get) // .get because isCoerceableFrom should make it safe
```

Comment:
yeah, I'm not really sure how to make this more idiomatic/function because the only generic way we have of coercing returns a Try.  Also the contract here is to throw an exception if it couldn't do the coercion so it's at least still following the contract.

Anybody else have any ideas on how to make this better?